modules/apps/27-interchain-accounts/types/packet.go

```diff
@@ -1,9 +1,40 @@
 package types
 
 import (
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
-func (packet InterchainAccountPacketData) GetBytes() []byte {
-	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&packet))
+// ValidateBasic performs basic validation of the interchain account packet data.
+// The memo may be empty.
+func (iapd InterchainAccountPacketData) ValidateBasic() error {
+	if iapd.Data == nil {
+		return sdkerrors.Wrap(ErrInvalidOutgoingData, "packet data cannot be empty")
+	}
+	// TODO: add type validation when data type enum supports unspecified type
```

Comment:
should we add type validation? Might be better not to for future compatibility purposes

Reply:
> I think there's only a single value at the moment which is somewhere between an unspecified and specified enum TYPE_EXECUTE_TX_UNSPECIFIED laughing

Thought I updated this in another PR :thinking: We should definitely update this.

Reply:
> This `ValidateBasic` happens on the `host` chain, right? Or is this validate function called manually?

Called on controller and host.

Reply:
Created #484